subprojects/base-services/src/main/java/org/gradle/internal/reflect/JavaReflectionUtil.java

```diff
@@ -244,12 +244,25 @@ public static Method findMethod(Class<?> target, Spec<Method> predicate) {
             handlerClass = fallbackType;
         }
         try {
-            return Cast.uncheckedCast(handlerClass.newInstance());
+            return Cast.uncheckedCast(handlerClass.getConstructor().newInstance());
```

Comment:
Shouldn't we use the `newInstance` method from just below here, too?

Reply:
If you search for all usage of this methods, you can see all classes loaded by this method is controlled by ourselves, so I choose `getConstructor().newInstance()`.